src/renderer/components/cluster-manager/cluster-status.tsx

```diff
@@ -94,58 +92,68 @@ export class ClusterStatus extends React.Component<Props> {
     }));
   };
 
-  renderContent() {
-    const { authOutput, cluster, hasErrors } = this;
-    const failureReason = cluster.failureReason;
+  renderAuthenticationOutput() {
+    return (
+      <pre>
+        {
+          this.authOutput.map(({ message, isError }, index) => (
+            <p key={index} className={cssNames({ error: isError })}>
+              {message.trim()}
+            </p>
+          ))
+        }
+      </pre>
+    );
+  }
 
-    if (!hasErrors || this.isReconnecting) {
-      return (
-        <div className="flex items-center column gaps">
-          <Spinner singleColor={false} className={styles.spinner} />
-          <pre className="kube-auth-out">
-            <p>{this.isReconnecting ? "Reconnecting" : "Connecting"}&hellip;</p>
-            {authOutput.map(({ data, error }, index) => {
-              return <p key={index} className={cssNames({ error })}>{data}</p>;
-            })}
-          </pre>
-        </div>
-      );
+  renderStatusIcon() {
+    if (this.hasErrors) {
+      return <Icon material="cloud_off" className={styles.icon} />;
     }
 
     return (
-      <div className="flex items-center column gaps">
-        <Icon material="cloud_off" className={styles.icon} />
-        <h2>
-          {cluster.preferences.clusterName}
-        </h2>
-        <pre>
-          {authOutput.map(({ data, error }, index) => {
-            return <p key={index} className={cssNames({ error })}>{data}</p>;
-          })}
+      <>
+        <Spinner singleColor={false} className={styles.spinner} />
+        <pre className="kube-auth-out">
+          <p>{this.isReconnecting ? "Reconnecting" : "Connecting"}&hellip;</p>
         </pre>
-        {failureReason && (
-          <div className="error">{failureReason}</div>
-        )}
-        <Button
-          primary
-          label="Reconnect"
-          className="box center"
-          onClick={this.reconnect}
-          waiting={this.isReconnecting}
-        />
-        <a
-          className="box center interactive"
-          onClick={this.manageProxySettings}>
-          Manage Proxy Settings
-        </a>
-      </div>
+      </>
     );
   }
 
+  renderReconnectionHelp() {
+    if (this.hasErrors && !this.isReconnecting) {
+      return (
+        <>
+          <Button
+            primary
+            label="Reconnect"
+            className="box center"
+            onClick={this.reconnect}
+            waiting={this.isReconnecting}
+          />
+          <a
+            className="box center interactive"
+            onClick={this.manageProxySettings}
+          >
+            Manage Proxy Settings
+          </a>
+        </>
+      );
+    }
+
+    return undefined;
```

Comment:
Why it's needed? Maybe just define to function with returning `:void`?

Reply:
Because the function doesn't just return void, it also returns a ReactNode